object_tracker/object_tracker.go

```diff
@@ -160,15 +167,19 @@ func (t *myTracker) run(stream gostream.VideoStream, cancelableCtx context.Conte
 			// Rename from temporal matches. New det copies old det's label
 			renamedNew := t.RenameFromMatches(matches, namedOld, filteredNew)
 
-			// Store the matched detections
+			// Store the matched detections and image
 			t.oldDetections.Store(&[2][]objdet.Detection{namedOld, renamedNew})
+			t.currImg.Store(&img)
 
-			done := time.Now()
-			took := done.Sub(start)
-			t.timeStats = append(t.timeStats, took)
+			took := time.Since(start)
 			waitFor := time.Duration((1/t.frequency)*float64(time.Second)) - took
 			if waitFor > time.Microsecond {
-				time.Sleep(waitFor)
+				select {
+				case <-cancelableCtx.Done():
+					return
+				case <-time.After(waitFor):
+					time.Sleep(waitFor)
```

Comment:
you don't want to sleep in addition to waiting with `time.After`, it will double the time you wait. You can remove the time.Sleep and just leave it blank.

Reply:
done